File: Medium/0781-Rabbits in Forest/HashMap.rb
# @param {Integer[]} answers
# @return {Integer}
def num_rabbits(answers)
    same_tell = Hash.new(0)
    ret = 0

    for answer in answers
        same_tell[answer + 1] += 1
    end

    same_tell.each do |k, v|
        ret += (v * 1.0 / k).ceil * k
    end

    return ret
end

File: Problemset/0781-Rabbits in Forest/HashMap.rs
use std::collections::HashMap;

impl Solution {
    pub fn num_rabbits(answers: Vec<i32>) -> i32 {
        let mut same_tell = HashMap::new();

        for answer in answers {
            *same_tell.entry(answer + 1).or_insert(0) += 1;
        }

        same_tell
            .iter()
            .map(|(&k, &v)| (v as f64 / k as f64).ceil() as i32 * k)
            .sum()
    }
}
